src/components/dialogs/dataSources/COVIDcast.svelte

```diff
@@ -23,38 +26,77 @@
     }
   }
 
-  onMount(() => {
-    fetchCOVIDcastMeta().then((res) => {
-      geoTypes = [...new Set(res.map((d) => d.geo_type))];
-      const byDataSource = new Map<string, LabelValue & { signals: string[] }>();
-      for (const row of res) {
-        const ds = byDataSource.get(row.data_source);
-        if (!ds) {
-          byDataSource.set(row.data_source, {
-            label: row.data_source,
-            value: row.data_source,
-            signals: [row.signal],
-          });
-        } else if (!ds.signals.includes(row.signal)) {
-          ds.signals.push(row.signal);
+  // Helper function; delay invoking "fn" until "ms" milliseconds have passed
+  const debounce = (fn: Function, ms = 500) => {
+    let timeoutId: ReturnType<typeof setTimeout>;
+    return function (this: any, ...args: any[]) {
+      clearTimeout(timeoutId);
+      timeoutId = setTimeout(() => fn.apply(this, args), ms);
+    };
+  };
+
+  function fetchMetadata() {
+    fetchCOVIDcastMeta(form_key).then((res) => {
+      if (res.length == 0) {
+        valid_key = form_key == ''; // mark key as valid if it's empty, otherwise invalid
```

Comment:
an empty response should come only from an invalid key; an empty key should actually be valid and thus it should never get an empty response.
```suggestion
        valid_key = false; // an empty response implies an invalid key
```